cmd/collector/collector.go

```diff
@@ -173,13 +181,42 @@ func run() error {
 		}
 	}()
 
+	// Create a new context with a cancel function
+	_, cancel := context.WithCancel(context.Background())
```

Comment:
you do not use the created context, that's definitely not right
if you do not need a cancellable context, don't create one

Reply:
Removed.